lib/apus/adapters/test_adapter.ex

```diff
@@ -15,7 +16,10 @@ defmodule Apus.TestAdapter do
   end
 
   def deliver(message, _config) do
-    send(self(), {:delivered_message, message})
+    case send(self(), {:delivered_message, message}) do
+      {:delivered_message, _} -> {:ok, message}
+      error -> {:error, error}
+    end
```

Comment:
The `send` function always returns the given input so we shouldn't need the `case` here.

As we can hardcode the response, maybe something like this would be clearer?
```elixir
send(self(), {:delivered_message, message})

{:ok, message}
```